release.py

```diff
@@ -0,0 +1,289 @@
+    if step_complete("install_deps_on_venv_complete"):
+        print(
+            f"{bcolors.OKCYAN}Skipping venv setup and dependency installation...{bcolors.ENDC}"
+        )
+    else:
+        try:
+            subprocess.run(["python", "-m", "venv", "venv"], check=True)
+            subprocess.run(
+                ["./venv/bin/pip", "install", "--editable", ".[test]"], check=True
+            )
+            record_step("install_deps_on_venv_complete")
+        except Exception as e:
+            print(
+                f"{bcolors.FAIL}INFO: Failed to install virtualenv and and dependencies clone! :(...{bcolors.ENDC}"
+            )
+            print(f"{bcolors.FAIL} {e} {bcolors.ENDC}")
+    if step_complete("branch_complete"):
+        print(f"{bcolors.OKCYAN}Skipping create release branch...{bcolors.ENDC}")
+    else:
+        try:
+            subprocess.run(["git", "branch", BRANCH], check=True)
+            record_step("branch_complete")
+        except Exception as e:
+            print(
+                f"{bcolors.FAIL}INFO: Failed to clean release branch! :(...{bcolors.ENDC}"
+            )
+            print(f"{bcolors.FAIL} {e} {bcolors.ENDC}")
+    subprocess.run(["git", "checkout", BRANCH])
+    if step_complete("tox_news_complete"):
+        print(f"{bcolors.OKCYAN}Skipping news generation...{bcolors.ENDC}")
+    else:
+        try:
+            subprocess.run(["./venv/bin/tox", "-e", "news"], check=True)
+            record_step("tox_news_complete")
+        except Exception as e:
+            print(f"{bcolors.FAIL}INFO: Failed to generate news! :(...{bcolors.ENDC}")
+            print(f"{bcolors.FAIL} {e} {bcolors.ENDC}")
+    if step_complete("newsfragment_complete"):
+        print(f"{bcolors.OKCYAN}Skipping add news fragment...{bcolors.ENDC}")
+    else:
+        try:
+            mkfile("newsfragments/{ticket}.minor".format(ticket=TICKET))
+            record_step("newsfragment_complete")
+        except Exception as e:
+            print(
+                f"{bcolors.FAIL}INFO: Failed to add newsfragment file! :(...{bcolors.ENDC}"
+            )
+            print(f"{bcolors.FAIL} {e} {bcolors.ENDC}")
+    if step_complete("commit_newsfragment_complete"):
+        print(f"{bcolors.OKCYAN}Skipping commit fragment...{bcolors.ENDC}")
+    else:
+        try:
+            subprocess.run(["git", "add", "."], check=True)
+            subprocess.run(
+                ["git", "commit", "-s", "-m", "tahoe-lafs-{tag} news".format(tag=TAG)],
+                check=True,
+            )
+            record_step("commit_newsfragment_complete")
+        except Exception as e:
+            print(
+                f"{bcolors.FAIL}INFO: Failed to commit newsfragment! :(...{bcolors.ENDC}"
+            )
+            print(f"{bcolors.FAIL} {e} {bcolors.ENDC}")
+    updated_content = None
+    LINE = "=" * len(RELEASE_TITLE)
+    with open("NEWS.rst", "r") as f:
+        content = f.read()
+        updated_content = re.sub(
```

Comment:
This doesn't end up doing the right thing; there is a newline missing. E.g. I get:

```Release 0.9999.1 (2022-05-05)=============================```

trying this now.